scipy/linalg/_solvers.py

```diff
@@ -80,6 +80,19 @@ def solve_sylvester(a, b, q):
     True
 
     """
+    for ind, _ in enumerate((a, b)):
+        if not np.equal(*_.shape):
+            raise ValueError("Matrix {} should be square.".format("ab"[ind]))
+    # Shape consistency check
+    if a.shape[0] != q.shape[0] or b.shape[0] != q.shape[1]:
+        raise ValueError("Matrix q should have shape (a.shape[0], b.shape[0])")
+
+    # Accomodate empty a
+    if a.size == 0 or b.size == 0:
+        dt = np.result_type(a, b, q)
+        if dt == int:
+            dt = np.float64
+        return np.empty(q.shape, dtype=dt)
```

Comment:
Here you can write the following

        tdict = {'s': np.float32, 'd': np.float64, 'c': np.complex64, 'z': np.complex128}
        func = get_lapack_funcs(('trsyl',), arrays=(a, b, q))
        return np.empty(q.shape, dtype=tdict[func.typecode])

First line creates the mapping from lapack types to numpy types, second line searches for the suitable type given the arrays (uses an example array of 'trsyl' but can be anything). Third line uses the resulting lapack type to get the numpy type of `q`.

Reply:
Thanks. Done 👍